stable/mysql/templates/deployment.yaml

```diff
@@ -91,6 +91,7 @@ spec:
         - name: MYSQL_ALLOW_EMPTY_PASSWORD
           value: "true"
         {{- end }}
+        {{- if not .Values.allowEmptyRootPassword }}
```

Comment:
I don't think we need an additional value for this. If `mysqlRootPassword` is not set and `mysqlAllowEmptyPassword` the mounted secret would be an empty string

Reply:
lgtm, fixed in this way. Thank you for a good idea.

This looks way more robust.